### DISCUSSION TITLE
Why baggage/trace are in one package but context is separated?

### DISCUSSION THREAD

**Discussion by bogdandrutu** [open-telemetry/opentelemetry-java#2625] [Q&A]

**Comment by anuraaga** [comment]
We wanted context to be independent of the OpenTelemetry API, potentially used outside it. Notably, by having it in a separate module, there isn't any way to add types like `Span` or `Baggage` to the context implementation, which I think is what people preferred.

**Reply by bogdandrutu** [reply]
My question was more like why baggage is not also a separate module?

**Reply by anuraaga** [reply]
I guess baggage seems like a telemetry feature as opposed to a utility but maybe not. We would like signals to be used together as `OpenTelemetry` so want the single artifact for exposing signals, which means metrics would also be merged in. It's true that baggage seems to not actually be a signal so it could probably be extracted if it felt better to do so.
